db/view/view.cc

```diff
@@ -1110,6 +1143,15 @@ future<stop_iteration> view_update_builder::on_results() {
                     auto update = clustering_row(existing.key(), row_tombstone(std::move(tombstone)), row_marker(), ::row());
                     generate_update(std::move(update), { std::move(existing) });
                 }
+            } else if (_existing->is_static_row()) {
+                auto existing = std::move(*_existing).as_static_row();
+                auto tombstone = _update_partition_tombstone;
+                // The static row might be unintentionally included when fetching existing clustering rows,
+                // even if the static row was not updated. In that case, the tombstone will be empty.
```

Comment:
Can you elaborate why this comment holds true? I'm not trying to undermine it, I just don't understand it - why would tombstone existence indicate that the static row was updated and vice versa? Is it guaranteed by some standard or is it just an implementation detail?

Reply:
AFAIK an existing static row can be affected either by:

1. A static row in the update mutation
2. A partition tombstone in the update mutation

If code execution reaches the comment, we can be sure that there is no (1). The `if` check that follows makes sure that we don't accidentally try to remove the row if there is no (2).

I'll update the comment with this information.